src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -371,6 +372,22 @@ private CSharpSyntaxNode ParseExpressionOrPattern(bool whenIsKeyword, Precedence
             }
         }
 
+        /// <summary>
+        /// Is the current token something that could follow a type in a pattern?
+        /// </summary>
+        private bool IsTokenFollowingTypeInPattern()
```

Comment:
The name seems to imply looking back to verify the previous tokens represent a type. Perhaps name `CanTokenFollowTypeInPattern` instead.

Reply:
Agreed.